### DISCUSSION TITLE
STEP Line graph

### DISCUSSION THREAD

**Discussion by Hammad2Kayani** [patrykandpatrick/vico#640] [Questions]
im currently displaying a curved graph, is there a way to convert this to a step line graph?

**Comment by Gowsky** [comment]
Hi, step charts are a planned addition. However, for certain use cases, you can create a custom implementation of `LineCartesianLayer.LineSpec.PointConnector` for `LineCartesianLayer` which would connect points with lines forming steps, instead of the default curved line.

**Comment by baum-sparton** [comment]
```
internal class StepPointConnector() :
    LineCartesianLayer.PointConnector {

    override fun connect(
        context: CartesianDrawingContext,
        path: Path,
        x1: Float,
        y1: Float,
        x2: Float,
        y2: Float,
    ) {
        path.lineTo(x2,y1)
        path.lineTo(x2,y2)
    }
}
```

**Reply by npced16** [reply]
Thank you! Your code made it easy to create the chart with straight lines.
